lib/presentation/redux/states/my_space_state.dart

```diff
@@ -35,34 +35,64 @@ import 'package:domain/domain.dart';
 import 'package:domain/src/state/failure.dart';
 import 'package:domain/src/state/success.dart';
 import 'package:flutter/foundation.dart';
+import 'package:linshare_flutter_app/presentation/model/file/selectable_element.dart';
 import 'package:linshare_flutter_app/presentation/redux/states/linshare_state.dart';
 
 @immutable
 class MySpaceState extends LinShareState {
-  final List<Document> documentList;
+  final List<SelectableElement<Document>> documentList;
+  final SelectMode selectMode;
 
-  MySpaceState(Either<Failure, Success> viewState, this.documentList) : super(viewState);
+  MySpaceState(
+    Either<Failure, Success> viewState,
+    this.documentList,
+    this.selectMode
+  ) : super(viewState);
 
   factory MySpaceState.initial() {
-    return MySpaceState(Right(IdleState()), []);
+    return MySpaceState(Right(IdleState()), [], SelectMode.INACTIVE);
   }
 
   @override
   LinShareState clearViewState() {
-    return MySpaceState(Right(IdleState()), []);
+    return MySpaceState(Right(IdleState()), documentList, selectMode);
   }
 
   @override
   LinShareState sendViewState({Either<Failure, Success> viewState}) {
-    return MySpaceState(viewState, documentList);
+    return MySpaceState(viewState, documentList, selectMode);
   }
 
   LinShareState setDocuments({Either<Failure, Success> viewState, List<Document> newDocumentList}) {
-    return MySpaceState(viewState, newDocumentList);
+    final selectedElements = documentList.where((element) => element.selectMode == SelectMode.ACTIVE).map((element) => element.element).toList();
+
+    return MySpaceState(viewState,
+      {for (var document in newDocumentList)
+          if (selectedElements.contains(document))
+            SelectableElement<Document>(document, SelectMode.ACTIVE)
+          else SelectableElement<Document>(document, SelectMode.INACTIVE)}.toList(),
+      selectMode);
+  }
```

Comment:
Please check the performance of the case: `refresh during selected mode`?. I concerned about when the list is rebuild.

Reply:
It doesn't bug or anything on Nexus API 30, have you seen smth ?

Reply:
No^^. So good, I will check in real device